crypto/rand/rand_vms.c

```diff
@@ -52,81 +47,71 @@ static struct items_data_st {
     {4, JPI$_PPGCNT},
     {4, JPI$_WSPEAK},
     {4, JPI$_FINALEXC},
-    {0, 0}                      /* zero terminated */
+    {0, 0}
 };
 
 int RAND_poll(void)
 {
-
     /* determine the number of items in the JPI array */
-
     struct items_data_st item_entry;
-    int item_entry_count = sizeof(items_data)/sizeof(item_entry);
-
+    int item_entry_count = OSSL_NELEM(items_data);
     /* Create the JPI itemlist array to hold item_data content */
-
     struct {
         short length, code;
         int *buffer;
         int *retlen;
-    } item[item_entry_count], *pitem; /* number of entries in items_data */
-
+    } item[item_entry_count], *pitem;
     struct items_data_st *pitems_data;
-    int data_buffer[(item_entry_count*2)+4]; /* 8 bytes per entry max */
+    int data_buffer[(item_entry_count * 2) + 4]; /* 8 bytes per entry max */
     int iosb[2];
     int sys_time[2];
     int *ptr;
     int i, j ;
     int tmp_length   = 0;
     int total_length = 0;
 
-    pitems_data = items_data;
-    pitem = item;
-
-
     /* Setup itemlist for GETJPI */
-    while (pitems_data->length) {
+    pitems_data = items_data;
+    for (pitem = item; pitems_data->length != 0; pitem++) {
         pitem->length = pitems_data->length;
         pitem->code   = pitems_data->code;
         pitem->buffer = &data_buffer[total_length];
         pitem->retlen = 0;
         /* total_length is in longwords */
-        total_length += pitems_data->length/4;
+        total_length += pitems_data->length / 4;
         pitems_data++;
-        pitem ++;
     }
     pitem->length = pitem->code = 0;
 
-    /* Fill data_buffer with various info bits from this process */
-    /* and twist that data to seed the SSL random number init    */
+    /*
+     * Fill data_buffer with various info bits from this process
+     * and twist that data to seed the SSL random number init
```

Comment:
It seems like the comment applies more to the for loop than the early-return conditional?